### ISSUE TITLE
`no-return-assign` error occured when using comma operator

### ISSUE THREAD

**Issue by wan2land** [eslint/eslint#13065]
**Tell us about your environment**

```
Environment Info:

Node version: v10.16.3
npm version: v6.9.0
Local ESLint version: v6.8.0 (Currently used)
Global ESLint version: Not found
```

**What parser (default, Babel-ESLint, etc.) are you using?**
no

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  env: {
    es6: true,
  },
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/* eslint no-return-assign: error */

;[].reduce((carry, { id, value }) => (carry[id] = value), {}) // safe
;[].reduce((carry, { id, value }) => (carry[id] = value, carry), {}) // error
```

```bash
npx eslint test.js
```

**What did you expect to happen?**

no error

**What actually happened? Please include the actual, raw output from ESLint.**

```
/Users/wan2land/Workspace/wan2land/eslint.info/test.js
  4:12  error  Arrow function should not return assignment  no-return-assign

✖ 1 problem (1 error, 0 warnings)
```

**Are you willing to submit a pull request to fix this bug?**

no

**Comment by wan2land** [comment]
i find same issue, https://github.com/eslint/eslint/issues/9418

results of the previous discussion made me sad.
